File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        x = []
        for i in list(s):
            if(i.isalnum()):
                x.append(i.lower())
        startPtr = 0
        endPtr = len(x) - 1
        while(startPtr < endPtr):
            if(x[startPtr] != x[endPtr]):
                return False
            startPtr += 1
            endPtr -= 1
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase().replaceAll("[^A-Za-z0-9]","");
        int startPtr = 0;
        int endPtr = s.length() - 1;

        while(startPtr < endPtr){
            if(s.charAt(startPtr) == s.charAt(endPtr)){
                startPtr++;
                endPtr--;
                continue;
            }
            return false;
        }

        return true;
    }
}